src/qililab/instrument_controllers/instrument_controllers.py

```diff
@@ -64,4 +65,4 @@ def __str__(self) -> str:
         Returns:
             str: String representation of the Instrument Controllers class.
         """
-        return str(yaml.dump(self.to_dict(), sort_keys=False))
+        return str(YAML().dump(self.to_dict(), io.BytesIO()))
```

Comment:
What is this new addition?

Reply:
`ruamel`, always needs a `stream`, unless you call it from a `context_manager`, so I just added default `stream` from `io` which I found in a stack exchange. 

`ruamel` also says this in their documentation, we could maybe use that:
![Screenshot 2023-10-25 at 19 28 11](https://github.com/qilimanjaro-tech/qililab/assets/109400222/fc92124d-2301-4ac2-98d9-3cba4d5b659c)

Reply:
Great!